query.go

```diff
@@ -638,6 +643,21 @@ checkResponses:
 	}
 }
 
+func (s *ChainService) getFilterFromCache(blockHash *chainhash.Hash,
+	filterType filterdb.FilterType) *gcs.Filter {
+	cacheKey := filterCacheKey{blockHash: blockHash, filterType: filterType}
+	if cachedFilter, ok := s.FilterCache.Get(cacheKey).(*gcs.Filter); ok {
```

Comment:
No need to check `ok`, can return the value directly I think.

Reply:
do you mean just:
`return s.FilterCache.Get(cacheKey).(*gcs.Filter)`
if I do that, the test fails with:
`panic: interface conversion: interface {} is nil, not *gcs.Filter`.